src/hazelcore/Monads.re

```diff
@@ -6,9 +6,39 @@
 
    In any case, that's a good reference. */
 
-module type MONAD = {
+module type MONAD_BASIC = {
   [@deriving sexp]
   type t('a);
   let return: 'a => t('a);
   let bind: (t('a), 'a => t('b)) => t('b);
+  let map: [ | `Define_using_bind | `Custom((t('a), 'a => 'b) => t('b))];
+};
+
+module type MONAD = {
+  include MONAD_BASIC;
+  let map: (t('a), 'a => 'b) => t('b);
+  let zip: (t('a), t('b)) => t(('a, 'b));
```

Comment:
I like to call this `zip` but I saw it was called `product` in another file. Is `zip` okay here?

Reply:
zip makes sense yeah